components/EditCoverSection.tsx

```diff
@@ -1,9 +1,271 @@
+
+  const handleUploadImageClick = () => {
+    if (hiddenFileInput?.current instanceof HTMLInputElement) {
+      hiddenFileInput.current.click();
+    }
+  };
+
+  const handleUpload = async (e: React.ChangeEvent) => {
+    const input = e.target as HTMLInputElement;
+    console.log(home.slogan);
+
+    const inputName = input.name;
+
+    if (!input.files?.length) {
+      return;
+    }
+    const file = input.files[0];
+
+    // display the image
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      setImageLink(e.target?.result as string);
+    };
+    reader.readAsDataURL(file);
+
+    let imageUrl;
+    try {
+      imageUrl = await uploadImage(file);
+      if (imageUrl) {
+        setHome({ ...home, [inputName]: imageUrl });
+      }
+    } catch (e) {
+      console.error("Image size is too large");
+      return;
+    }
+  };
+
+  const handleView = () => {
+    //
+  };
+
   return (
-    <div>EditCoverInfoSection</div>
-  )
-}
+    <section className="contact">
+      <div className="container ml-24 px-8 py-4 md:px-16">
+        <h2 className="mt-12 mb-4 text-xl">home</h2>
+        <h3 className="mb-12">Edit cover information</h3>
+        <div className="mb-4 grid gap-8 md:grid-cols-2 [&_td]:p-2">
+          <div className="col-1">
+            <div className="h-300 relative mb-4 flex">
+              <Image
+                width="500"
+                height="200"
+                src={cover || "/asset1.png"}
+                style={{
+                  objectFit: "cover",
+                }}
+                alt="cover"
+              />
+              <div
+                className="cover-edit-icon"
+                onClick={() => handleOpen("cover")}
+              >
+                <ReactSVG src="/edit.svg" className="edit-svg h-3 w-3" />
+              </div>
+              {isOpen === "cover" && (
+                <div className="cover-edit-container">
+                  <label htmlFor="image">
+                    <button onClick={handleView} className="cover-edit-btn">
+                      view photo
+                    </button>
+                  </label>
+                  <label htmlFor="cover">
+                    <button
+                      onClick={handleUploadImageClick}
+                      className="cover-edit-btn"
+                    >
+                      upload a photo
+                    </button>
+                  </label>
+                  <input
+                    name="cover"
+                    ref={hiddenFileInput}
+                    type="file"
+                    className="hidden"
+                    onChange={handleUpload}
+                  />
+                </div>
+              )}
+            </div>
+          </div>
+          <div className="col-2">
+            <div className="grid gap-4 md:grid-cols-4">
+              <label>Edit text</label>
+              <textarea
+                className="input input-primary col-span-3"
```

Comment:
in className no need to say "input input-primary"
Make "input-primary" components apply "input" from css, example:
  .input-primary{ @apply input ... }